test/type/subclass_composition.swift

```diff
@@ -126,8 +126,8 @@ func basicSubtyping(
   let _: Derived & AnyObject = derived
 
   let _ = base as Base<Int> & P1
-  let _ = base as Base<Int> & AnyObject
-  let _ = derived as Derived & AnyObject
+  let _ = base as Base<Int> & AnyObject // expected-warning {{casting expression to 'Base<Int>' doesn't change the type}} {{16-41=}}
```

Comment:
Those two here, I'm not sure if we should show all the type coercion type in the message like 'casting expression to 'Base<Int> & AnyObject' doesn't change the type'
cc @xedin